pallets/circuit/src/lib.rs

```diff
@@ -1297,10 +1317,15 @@ impl<T: Config> Pallet<T> {
                     local_ctx.xtx_id,
                     local_ctx.xtx.timeouts_at,
                 );
-                <PendingXtxBidsTimeoutsMap<T>>::insert::<XExecSignalId<T>, T::BlockNumber>(
-                    local_ctx.xtx_id,
-                    T::SFXBiddingPeriod::get() + frame_system::Pallet::<T>::block_number(),
-                );
+                // cannot return an error if checked_add fails
+                if let Some(v) = T::SFXBiddingPeriod::get()
```

Comment:
same here

Reply:
How can I short-circuit in this case where I have a fn with return type Weight? Should I change it or just log::error!() and skip the loop?

Reply:
just log and return the current weight counter, or you could break the loop ya